src/components/ShowList.jsx

```diff
@@ -31,11 +43,38 @@ function ShowList() {
       )}
       {value && value.docs.length > 0 && (
         <div>
-          Collection:
+          <input
+            type="text"
+            placeholder="Filter items..."
+            value={filter}
+            onChange={handleChange}
+          />
+          {filter && (
+            <button
+              onClick={() => setFilter('')}
+              aria-label="clear filter text"
+            >
+              <FontAwesomeIcon icon={['fas', 'times']} />
+            </button>
+          )}
           <ul>
-            {value.docs.map((doc) => (
-              <Item key={doc.id} {...doc.data()} id={doc.id} />
-            ))}
+            {filter
+              ? value.docs
+                  .map((doc) => ({ id: doc.id, ...doc.data() }))
+                  .filter((item) =>
+                    item.name.toLowerCase().includes(filter.toLowerCase()),
+                  )
+                  .map((item) => (
+                    <Item
+                      key={item.id}
+                      id={doc.id}
+                      lastPurchasedDate={item.lastPurchasedDate}
+                      name={item.name}
+                    />
+                  ))
+              : value.docs.map((doc) => (
+                  <Item key={doc.id} {...doc.data()} id={doc.id} />
```

Comment:
I see that we are rendering the `Item` component twice. Can you please let me know why does it happen? Thank you! 🙌

Reply:
When filter is an empty string, we render all value.docs objects as Item components as we're assuming the user wants to see the full list 😊